layerindex/utils.py

```diff
@@ -253,33 +290,53 @@ def parse_layer_conf(layerdir, data, logger=None):
     data = parse_conf(conf_file, data)
     data.expandVarref('LAYERDIR')
 
-def runcmd(cmd, destdir=None, printerr=True, logger=None):
+child_pid = 0
+def runcmd(cmd, destdir=None, printerr=True, outfile=None, logger=None, shell=False):
     """
         execute command, raise CalledProcessError if fail
         return output if succeed
     """
     if logger:
         logger.debug("run cmd '%s' in %s" % (cmd, os.getcwd() if destdir is None else destdir))
-    out = tempfile.TemporaryFile()
+    if outfile:
+        out = open(outfile, 'wb+')
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *out*. The resource is allocated by call *builtins.open*. Currently, there are execution paths that do not contain closure statements, for example, when *signal()* throws an exception. To prevent this resource leak, close *out* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).